src/lib/util/Notifications/Twitch.ts

```diff
@@ -0,0 +1,149 @@
+import { createHmac } from 'crypto';
+import { TOKENS, TWITCH_CALLBACK } from '../../../../config';
+import { fetch, enumerable, FetchResultTypes } from '../util';
+import { Mime, Time } from '../constants';
+import { TwitchKrakenChannelSearchResults, TwitchHelixResponse, TwitchHelixGameSearchResult, TwitchHelixBearerToken, TwitchHelixUsersSearchResult } from '../../types/definitions/Twitch';
+import { RateLimitManager } from 'klasa';
+
+export const enum ApiVersion {
+	Kraken,
+	Helix
+}
+
+export const enum TwitchHooksAction {
+	Subscribe = 'subscribe',
+	Unsubscribe = 'unsubscribe'
+}
+
+export interface OauthResponse {
+	access_token: string;
+	refresh_token: string;
+	scope: string;
+	expires_in: number;
+}
+
+export class Twitch {
+
+	public readonly ratelimitsStreams = new RateLimitManager(1, Twitch.RATELIMIT_COOLDOWN);
+	public readonly BASE_URL_HELIX = 'https://api.twitch.tv/helix/';
```

Comment:
Shouldn't this use constants?

Reply:
While most of the code uses constants, the newest code has been migrating to moving them as readonly instance fields, which makes them easier to inspect when debugging or even, to modify (with eval).